packages/payment-plugin/src/mollie/mollie.handler.ts

```diff
@@ -0,0 +1,123 @@
+
+let paymentMethodService: PaymentMethodService;
+let options: MolliePluginOptions;
+export const molliePaymentHandler = new PaymentMethodHandler({
+    code: 'mollie-payment-handler',
+    description: [
+        {
+            languageCode: LanguageCode.en,
+            value: 'Mollie payment',
+        },
+    ],
+    args: {
+        apiKey: {
+            type: 'string',
+        },
+        redirectUrl: {
+            type: 'string',
+        },
+    },
+    init(injector) {
+        paymentMethodService = injector.get(PaymentMethodService);
+        options = injector.get(PLUGIN_INIT_OPTIONS);
+    },
+    createPayment: async (ctx, order, amount, args, _metadata): Promise<CreatePaymentResult> => {
+        try {
+            const { apiKey } = args;
+            let { redirectUrl } = args;
+            const paymentMethods = await paymentMethodService.findAll(ctx);
+            const paymentMethod = paymentMethods.items.find(pm =>
+                pm.handler.args.find(arg => arg.value === apiKey) && pm.handler.args.find(arg => arg.value === redirectUrl),
+            );
+            if (!paymentMethod) {
+                throw Error(`No paymentMethod found for given apiKey`); // This should never happen
+            }
+            const mollieClient = createMollieClient({ apiKey });
+            redirectUrl = redirectUrl.endsWith('/') ? redirectUrl.slice(0, -1) : redirectUrl; // remove appending slash
+            const vendureHost = options.vendureHost.endsWith('/')
+                ? options.vendureHost.slice(0, -1)
+                : options.vendureHost; // remove appending slash
+            const payment = await mollieClient.payments.create({
+                amount: {
+                    value: `${(order.totalWithTax / 100).toFixed(2)}`,
+                    currency: order.currencyCode,
+                },
+                metadata: {
+                    orderCode: order.code,
+                },
+                description: `Order ${order.code}`,
+                redirectUrl: `${redirectUrl}/${order.code}`,
+                webhookUrl: `${vendureHost}/payments/mollie/${ctx.channel.token}/${paymentMethod.id}`,
+            });
+            return {
+                amount: order.totalWithTax,
+                transactionId: payment.id,
+                state: 'Authorized' as const,
+                metadata: {
+                    public: {
+                        redirectLink: payment.getPaymentUrl(),
+                    },
+                },
+            };
+        } catch (err) {
+            Logger.error(err, loggerCtx);
+            return {
+                amount: order.totalWithTax,
+                state: 'Declined' as const,
+                metadata: {
+                    errorMessage: err.message,
+                },
+            };
+        }
+    },
+    settlePayment: async (order, payment, args): Promise<SettlePaymentResult> => {
+        return { success: true };
```

Comment:
Could add a comment here that "Logic for settlement is handled by the webhook in mollie.controller.ts"